forge/ee/routes/staticAssets/index.js

```diff
@@ -0,0 +1,242 @@
+                    }
+                }
+            })
+            reply.send(result)
+        } catch (err) {
+            if (err.statusCode === 404) {
+                reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+            } else {
+                reply.code(400).send({ code: 'invalid_request', error: err.toString() })
+            }
+        }
+    })
+
+    app.put('/_/*', {
+        preHandler: app.needsPermission('project:files:edit'),
+        schema: {
+            summary: 'Update file properties in the instance',
+            tags: ['Instance Files'],
+            params: {
+                type: 'object',
+                properties: {
+                    instanceId: { type: 'string' }
+                }
+            },
+            body: {
+                type: 'object',
+                properties: {
+                    path: { type: 'string' },
+                    share: { type: 'object', additionalProperties: true }
+                }
+            }
+            // TODO: response format schema
+        }
+    }, async (request, reply) => {
+        const rootPath = `/api/v1/projects/${request.project.id}/files/_/`
+        let filePath = request.url.substring(rootPath.length)
+        if (/\/$/.test(filePath)) {
+            filePath = filePath.substring(0, filePath.length - 1)
+        }
+        const update = request.body
+
+        if (update.path && update.share) {
+            reply.code(400).send({ code: 'invalid_request', error: 'Cannot modify path and share in one request' })
+            return
+        }
+        if (update.path) {
+            try {
+                const result = await app.containers.updateFile(request.project, filePath, update)
+                reply.send(result)
+            } catch (err) {
+                if (err.statusCode === 404) {
+                    reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+                } else {
+                    reply.code(400).send({ code: 'invalid_request', error: err.toString() })
+                }
+            }
+        } else if (update.share) {
+            // Need to validate this is a directory, not a file
+            try {
+                await app.containers.listFiles(request.project, filePath)
+                // This is a valid directory so we can update the share details
+                const sharingConfig = await request.project.getSetting(KEY_SHARED_ASSETS) || {}
+                // TODO: validate contents of update.share
+                sharingConfig[filePath] = update.share
+                await request.project.updateSetting(KEY_SHARED_ASSETS, sharingConfig)
+            } catch (err) {
+                // Not a directory or 404
+                reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+                return
+            }
+
+            reply.code(200).send({})
+        }
+    })
+
+    app.delete('/_/*', {
```

Comment:
I am not comfortable with the API paths permitting `*`.  I think, as the original issue stated, this should be a named parameter 

> DELETE /api/v1/projects/:instanceId/files/_/:path

As @hardillb discovered in https://github.com/FlowFuse/flowfuse/pull/4392#issuecomment-2304401935, fastify resolves URLs before routing.  i.e. a call to `api/v1/projects/ba2bbf57-e491-46c4-a3a5-41adbd3be413/files/_/../../` would be resolved to `api/v1/projects/ba2bbf57-e491-46c4-a3a5-41adbd3be413` and in theory, delete the instance.

My recommendation is that file path value is URL encoded and treated as a single parameter (on all routes)

Reply:
Fixed by URI encoding the path